File: QuickStarts/CS/EventBroker/Tests/Tests.EventBrokerExtension/TestObjects/ClipboardManager.cs
//===============================================================================
// Microsoft patterns & practices
// Unity Application Block
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;
using System.Collections.Generic;
using System.Text;
using SimpleEventBroker;

namespace Tests.EventBrokerExtension.TestObjects
{
    class ClipboardManager
    {
        // turn off "event is never used" warning
        #pragma warning disable 67

        [Publishes("cut")]
        public event EventHandler Cut;

        [Publishes("copy")]
        public event EventHandler Copy;

        [Publishes("paste")]
        public event EventHandler Paste;

        #pragma warning restore 67

        [SubscribesTo("copy")]
        public void OnCopy(object sender, EventArgs e)
        {
        }

        [SubscribesTo("clipboard data available")]
        public void OnClipboardDataAvailable(object sender, EventArgs e)
        {
            
        }
    }
}

File: quickstarts/VB/EventBroker/Tests/Tests.EventBrokerExtension/TestObjects/ClipboardManager.vb
' Copyright (c) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports SimpleEventBroker

Namespace Tests.EventBrokerExtension.TestObjects
	Class ClipboardManager
		' turn off "event is never used" warning
    <Publishes("cut")> _
  Public Event Cut As EventHandler

		<Publishes("copy")> _
		Public Event Copy As EventHandler

		<Publishes("paste")> _
		Public Event Paste As EventHandler


		<SubscribesTo("copy")> _
		Public Sub OnCopy(ByVal sender As Object, ByVal e As EventArgs)
		End Sub

		<SubscribesTo("clipboard data available")> _
		Public Sub OnClipboardDataAvailable(ByVal sender As Object, ByVal e As EventArgs)

		End Sub
	End Class
End Namespace
